### PR TITLE
AccessData API Update + Unit-tests (New Branch)

### PR DESCRIPTION
**Pull request by ThomasVieth**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
Rewrite the current integration to provide interaction the new FTK API.

## Screenshots
AccessData__Process_Evidence_and_Export_Wed_Dec_29_2021
AccessData__Job_Polling_Tue_Jan_11_2022
image

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

**Comment by ThomasVieth** [comment]
Please await unit-test file. This will be provided as well today.

**Comment by kgal-pan** [comment]
> Please await unit-test file. This will be provided as well today.

@ThomasVieth  Looks like the UTs are fine. The build is failing for other reasons, namely:

- [Commands mentioned in YAML but not in Python code](https://app.circleci.com/pipelines/github/demisto/content/148179/workflows/1e9e7581-0465-4d98-b00c-6d3b535360f5/jobs/488298?invite=true#step-107-56).

- [Missing `test-module` command](https://app.circleci.com/pipelines/github/demisto/content/148179/workflows/1e9e7581-0465-4d98-b00c-6d3b535360f5/jobs/488298?invite=true#step-107-57). See https://xsoar.pan.dev/docs/integrations/code-conventions#test-module for details how to implement.

- [Playbooks formatting issues](https://app.circleci.com/pipelines/github/demisto/content/148179/workflows/1e9e7581-0465-4d98-b00c-6d3b535360f5/jobs/488299?invite=true#step-108-92). Run the following command to fix them:

```bash
demisto-sdk format -i Packs/Accessdata/Playbooks/AccessDataProcessEvidenceAndExport.yml
demisto-sdk format -i Packs/Accessdata/Playbooks/AccessDataJobPolling.yml
```

- [Integration images are not in expected dimensions](https://app.circleci.com/pipelines/github/demisto/content/148179/workflows/1e9e7581-0465-4d98-b00c-6d3b535360f5/jobs/488299?invite=true#step-108-109). 



In addition, please [sign the Contribution Form](https://forms.gle/XDfxU4E61ZwEESSMA).

**Comment by bziser** [comment]
@ThomasVieth Thanks for your contribution!
Looks great.
Can you fix the problems which the [validate](https://app.circleci.com/pipelines/github/demisto/content/148403/workflows/18bdcf8c-25ff-4d7b-a7cd-838718720fd5/jobs/488664) and [lint](https://app.circleci.com/pipelines/github/demisto/content/148403/workflows/18bdcf8c-25ff-4d7b-a7cd-838718720fd5/jobs/488663/steps) are failing for? Also Merge from master to update the pr. 
Thanks!

**Comment by kgal-pan** [comment]
@ThomasVieth - Looks like the last commit merge added unwanted commits to this PR. Please revert to commit [`58ccf52`](https://github.com/demisto/content/pull/17916/commits/58ccf525059b76fd69fc5bb610a100c7d0ff32bc) and use the script below to sync your fork with `master`:

```bash
./Utils/git_pull_master_into_fork.sh
```

**Comment by kgal-pan** [comment]
@ThomasVieth - Seeing [some issues running the unit tests](https://app.circleci.com/pipelines/github/demisto/content/148493/workflows/24e5a8b1-c0e8-4778-b74d-db563b16c759/jobs/488917). Can you look into this?

**Comment by ThomasVieth** [comment]
> @ThomasVieth - Seeing [some issues running the unit tests](https://app.circleci.com/pipelines/github/demisto/content/148493/workflows/24e5a8b1-c0e8-4778-b74d-db563b16c759/jobs/488917). Can you look into this?

Thanks, am taking a look today 👍 May require a docker image update due to this is the logging mechanism failing to retrieve the user information to log against.

**Comment by kgal-pan** [comment]
> Thanks, am taking a look today 👍 May require a docker image update due to this is the logging mechanism failing to retrieve the user information to log against.

@ThomasVieth - Any updates? I don't see any PRs in https://github.com/demisto/dockerfiles/pulls?q=is%3Apr+sort%3Aupdated-desc+accessdata+is%3Aopen

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  type: 0
+  required: true
+- display: Service Listening Port
+  name: PORT
+  defaultvalue: 4443
+  type: 0
+  required: true
+- display: "The API authentication key. http://support.accessdata.com/hc/en-us/articles/360053994573-Generating-API-Keys"
+  name: APIKEY
+  type: 0
```
Comment (kgal-pan) [inline Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml:28]: Change to `type: 4`, this means encrypted text block.

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  required: true
+  hidden: true
```
[the remark this answers is not in the document]
Comment (ThomasVieth) [reply Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml]: Cheers, field has been removed.

**Comment by kgal-pan** [comment]
@ThomasVieth - 

- Please update your fork using:

```bash
./Utils/git_pull_master_into_fork.sh
```

- Also, still waiting for resolution of UT import failures from client module. You mentioned something about a need to update the Docker image. Do we have a PR for that open?

**Comment by kgal-pan** [comment]
@ThomasVieth - Any updates? We might need to close this due to inactivity.

**Pull request closed without merging**